js/search/domain.js

```diff
@@ -217,23 +180,55 @@ const Domain = new Lang.Class({
     },
 
     /**
-     * Function: load_sync
+     * Function: load_shards_sync
      *
      * Loads up the shard needed for this domain.
      */
-    load_sync: function () {
-        if (this._shard_inited)
+    load_shards_sync: function () {
+        if (this._shards_inited)
             return;
 
-        this.get_shards(null);
+        let manifest_file = this._get_manifest_file();
+        // If the manifest.json doesn't exist, and we have a manifest in the bundle, symlink
+        // to it to bootstrap our subscription.
+        if (!manifest_file.query_exists(null)) {
+            let bundle_dir = this._get_bundle_dir();
+            let bundle_manifest_file = bundle_dir.get_child('manifest.json');
+            if (bundle_manifest_file.query_exists(null)) {
+                manifest_file.make_symbolic_link(bundle_manifest_file.get_path(), null);
+            } else {
+                throw new Gio.IOErrorEnum({
+                    message: 'You have no manifest.json and are not ' +
+                        'running from a Flatpak bundle. You must download' +
+                        ' a content update.',
+                    code: Gio.IOErrorEnum.NOT_FOUND,
+                });
+            }
+        }
+
+        this._make_bundle_symlinks(null);
+
+        let [success, data] = manifest_file.load_contents(null);
+        let manifest = JSON.parse(data);
+
+        let subscription_dir = this._get_subscription_dir();
+        this._shards = manifest.shards.map(function(shard_entry) {
+            let file = subscription_dir.get_child(shard_entry.path);
+            return new EosShard.ShardFile({
+                path: file.get_path(),
+            });
+        });
         // Don't allow init() to be cancelled; otherwise,
         // cancellation will spoil the object for future use.
         Eknc.utils_parallel_init(this._shards, 0, null);
 
+        // Append shards to default EknVfs for ekn:// uri to work
+        Eknc.default_vfs_set_shards(this._shards);
```

Comment:
There are some cases, like the xapian DB or link tables, where we explicitly have SHA collisions between shards. We don't load those from the VFS, so it should be fine, but it feels weird to me to do that.